src/client/features/interactions/interactions-edge-tooltip.js

```diff
@@ -24,11 +33,22 @@ class InteractionsEdgeTooltip extends React.Component {
   }
 
   render(){
+    const state = this.state;
     let { edge } = this.props;
     let { publications } = this.state;
     let title = edge.data('id');
+    let providers = edge.data('providers');
     let pcIds = edge.data('pcIds');
 
+    let providersList = providers.map( provider => {
+      const dsInfo = _.isEmpty(state.dataSources) ? {}: _.find(state.dataSources, ds => {
```

Comment:
I don't like introducing another instance of the 'find datasource by id' problem into another area of the codebase.  We need to come up with a solution to keep this logic in one place and make it reusable.